### PR TITLE
Clarify docs for got.HTTPError

### LINKED ISSUE #1213: Note that 304 doesn't throw `HTTPError`
#### What would you like to discuss?

The PR #252 updated `got` not to throw `HTTPErrors`. The ReadMe was updated that in case of 304 response status code the body would be empty. However the documentation on `HTTPError` still says that they are thrown for non-20x codes. 

I read the documentation on `HTTPError` and expected an exception for 304. It didn’t work this way. Would it make sense to update this part?

#### Checklist

- [x] I have read the documentation.

### PR DESCRIPTION
**Pull request by chinesedfan**
#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [ ] I have included some tests.
- [ ] If it's a new feature, I have included documentation updates.

Closed #1213. The preview is as following, and note that the anchor is a relative link so it doesn't work here but doesn't matter in fact.

----

Determines if a [`got.HTTPError`](#gothttperror) is thrown for error responses.

#### got.ParseError

> *ok*: 2xx or 304, and includes other 3xx if `options.followRedirect` is set

When server response code is ***ok***, and parsing body fails. Includes a `response` property.

#### got.HTTPError

When the server response code is not ***ok***. Includes a `response` property.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -523,7 +523,7 @@ const got = require('got');
 Type: `boolean`\
 Default: `true`
 
-Determines if a `got.HTTPError` is thrown for error responses (non-2xx status codes).
+Determines if a [`got.HTTPError`](#gothttperror) is thrown for error responses.
```
Comment (szmarczak) [inline readme.md:526]:
```suggestion
Determines if a [`got.HTTPError`](#gothttperror) is thrown for unsuccessful responses.
```

File: readme.md [hunk 2]
```diff
@@ -1298,11 +1298,13 @@ When reading from response stream fails.
 
 #### got.ParseError
 
-When server response code is 2xx, and parsing body fails. Includes a `response` property.
+> *ok*: 2xx or 304, and includes other 3xx if `options.followRedirect` is set
+
+When server response code is ***ok***, and parsing body fails. Includes a `response` property.
```
Comment (szmarczak) [inline readme.md:1303]: This actually does not affect `ParseError`.

File: readme.md (hunk 1 above)
Comment (szmarczak) [inline readme.md:526]:
```suggestion
Determines if a [`got.HTTPError`](#gothttperror) is thrown for unsuccessful responses.
```

File: readme.md [hunk 3]
```diff
@@ -1298,11 +1298,13 @@ When reading from response stream fails.
...
 
 #### got.HTTPError
 
-When the server response code is not 2xx. Includes a `response` property.
+When the server response code is not ***ok***. Includes a `response` property.
```
Comment (szmarczak) [inline readme.md:1307]: You need to replace `not ok` with a more descriptive information.

File: readme.md (hunk 2 above)
Comment (chinesedfan) [reply readme.md:1303]: @szmarczak Not really. See source/as-promise/index.ts.

```
// Parse body
try {
    response.body = parseBody(response, options.responseType, options.encoding);
} catch (error) {
    // Fallback to `utf8`
    response.body = rawBody.toString();

    if (isOk()) {
        // TODO: Call `request._beforeError`, see https://github.com/nodejs/node/issues/32995
        reject(error);
        return;
    }
}
```

File: readme.md (hunk 1 above)
Comment (chinesedfan) [reply readme.md]: Same with above. Looks like a duplicated suggestion.

File: readme.md (hunk 3 above)
Comment (chinesedfan) [reply readme.md:1307]: I want to make `ok` as a term, as well as used in `ParseError`.

File: readme.md (hunk 2 above)
Comment (szmarczak) [reply readme.md:1303]: It doesn't have any effect. This code is **not** executed on redirects. `304` are empty responses, so even with `responseType: 'json'` it never throws:

https://github.com/sindresorhus/got/blob/ece94edec31bc8d004f93d74e218d8fd3b442c15/source/as-promise/core.ts#L28-L30

There is no `response` event emitted on redirect:

https://github.com/sindresorhus/got/blob/ece94edec31bc8d004f93d74e218d8fd3b442c15/source/core/index.ts#L1032-L1114

File: readme.md (hunk 3 above)
Comment (szmarczak) [reply readme.md:1307]: Keep things simple. Don't redefine things. Simply:

> When the server response code is not 2xx and 3xx if `options.followRedirect` is `true`.

**Comment by szmarczak** [comment]
Looks good. Thanks for submitting! 🙌

**Pull request merged**